pandas/tests/indexes/multi/test_join.py

```diff
@@ -87,3 +87,20 @@ def test_join_self_unique(idx, join_type):
     if idx.is_unique:
         joined = idx.join(idx, how=join_type)
         assert (idx == joined).all()
+
+
+def test_join_multi_wrong_order():
+    # GH 25760
+    # GH 28956
+
+    midx1 = pd.MultiIndex.from_product([[1, 2], [3, 4]], names=["a", "b"])
+    midx2 = pd.MultiIndex.from_product([[1, 2], [3, 4]], names=["b", "a"])
+
+    join_idx, lidx, ridx = midx1.join(midx2, return_indexers=False)
+
+    exp_ridx = np.array([-1, -1, -1, -1], dtype=np.intp)
+
+    assert midx1.equals(join_idx)
+    assert midx2.equals(join_idx)
+    assert lidx is None
+    tm.assert_numpy_array_equal(ridx, exp_ridx)
```

Comment:
Can you use tm.assert_index_equal here instead of breaking up into individual arrays?

Reply:
I can only do this for midx1 as Index.equals is not sensitive for the order of names, but tm.assert_index_equal is.

Anyway, I expanded both the test and the docs.

Reply:
mdx does not equal join_idx, so midx2 just remains the same, basically, which is fine for Index.equals, but there is nothing to compare it with in this example using tm.assert_index_equal. That would require a more complicated example. I didn't add any, as the order of the indices is not agreed upon, I just corrected a bug one way.

Reply:
anyway, since this I expanded the test a little